### ISSUE TITLE
Create CSV file for monthly Birthday Card Recipients 

### ISSUE THREAD

**Issue by FrogGirl1123** [episphere/connect#831]
Starting a new issue separate from #736 as the API will be developed in the future

As decided through email and Punch List conversations on Wed., Nov. 29, we will put a hold on development of the API filter. Instead we will create a csv file with the with the following requirements to be shared with NORC via Box as we do for the BQ1 data for the sites:
For all participants who are verified and not withdrawn and have birthdays in a given month (e.g. January for out first shared file).
The csv file will contain:
Connect ID
Name (current first and last name from User Profile)
Mailing address (current address from User Profile)

@brotzmanmj please add any additional requirements and let @mnataraj92 where to set up the folder for NORC. Also what would we like as the delivery date, first day of the month, end of the previous month?

@brotzmanmj, @sonyekere, @anthonypetersen should we break this into a new Issue since the API work will resume eventually?

@jeannewu I would like you to write the scheduled query for this table and @jacobmpeters to work on getting the resulting csv into Box

Updates from replies in #736:
From @brotzmanmj add "not deceased" as a requirement

**Comment by jeannewu** [comment]
@FrogGirl1123 @brotzmanmj **may you please give me a folder in the box to save a copy of the birthday card csv file if possible?** Thanks

**Comment by jeannewu** [comment]
@FrogGirl1123 @brotzmanmj may you please assign a folder in Box to test and review the birthday card csv file if possible? Thanks a lot

**Comment by jacobmpeters** [comment]
Do we want the name and address to be in a single field as a printable address label?

Example from stage:

<img width="530" alt="image" src="https://github.com/episphere/connect/assets/7967451/5fffd8e2-4108-4d0e-a5f8-ececd634dea4">

**Comment by jeannewu** [comment]
@jacobmpeters Nice! Thanks a lot.

**Comment by jeannewu** [comment]
Hi, Jake, the csv file name with Madhuri's suggestion would be like "**CSV file for ",month," Birthday Card Recipients_2024.csv**". If possible, would you like to help me update the name of the csv file? Thanks a lot.

**Comment by jacobmpeters** [comment]
Hi @jeannewu. Thank you for letting me know. I wasn't aware of this requirement. I can update the CSV name. I will replace the spaces with underscores.

**Comment by jacobmpeters** [comment]
Here is an updated CSV file with an updated name: https://nih.app.box.com/file/1379431144150

Filename: CSV_file_for_December_2023_birthday_card_recipients_000000000000.csv

Note: The "_000000000000" is added by BigQuery. If the file size is too large, BigQuery will automatically clip the data into multiple CSV files and increment this tag by 1 for each additional file produced. We should not remove it until we know how this scales in production.

**Comment by jacobmpeters** [comment]
Jing mentioned that Madhuri and/or the Operations team would not to have spaces in the filename. If this is the case, it is possible to do this:

"CSV file for December 2023 Birthday Card Recipients_000000000000.csv"
https://nih.app.box.com/file/1379437544841

Generally speaking it's not good practice to have spaces in filenames because different programs and operating systems handle spaces differently. It happens to work for my pipeline, but it might not work for the users'.

**Comment by mnataraj92** [comment]
@jacobmpeters i think in that case it's fine to have "_" separating the words in the filename!

**Comment by jacobmpeters** [comment]
@mnataraj92 Ok! Sounds good. I'll switch it back :)

**Comment by mnataraj92** [comment]
It should just be "CSV_file_for_[insert month here]_Birthday_Card_Recipients_2024" so January would be "CSV_file_for_January_Birthday_Card_Recipients_2024"

Thanks!!

**Comment by jeannewu** [comment]
Thank you very much for your helpful comments. And right now, the code is set up at the corresponding month of each Birth month to pull out the list. Please let us know if the list would be prepared in advance (say one month earlier). It will be helpful  for us to update the script. Thanks a lot.

**Comment by jeannewu** [comment]
@FrogGirl1123 I talked with you this afternoon about the preparation for the January list of Birthday card. As we agreed, we might get the list one month earlier for the NORC birthday participants in the coming month, right?

**Comment by jacobmpeters** [comment]
> @FrogGirl1123 I talked with you this afternoon about the preparation for the January list of Birthday card. As we agreed, we might get the list one month earlier for the NORC birthday participants in the coming month, right?

Hi @jeannewu and @FrogGirl1123, I incorperated Jing's suggestion. The query now exports next month's birthday list:

[https://nih.app.box.com/file/1379497441479](https://nih.app.box.com/file/1379497441479)

**Comment by mnataraj92** [comment]
Hi @jacobmpeters and @jeannewu 

Here is what NORC said in regards to how they'd like the addresses & when they would like the csv file by

1)	Please send as columns: first_name, last_name, address_line_1, address_line_2, city, and zip_code
2)	To get the cards out early each month, please provide the CSV files by the 22nd of each prior month 
             a.	For example, we’ll need the January file by December 22nd

**Comment by jacobmpeters** [comment]
@mnataraj92 Thanks for letting us know. This is very clear and doable!

I would like to set up an additional test folder for the production data tomorrow so that we can see how the CSV delivery scales up before we share to the NORC folder. 

Could you please make a test folder for prod and invite Jing and me as "editors"? Could you also add AutomationUser_1960507_dBBoRFHiim@boxdevedition.com as a "viewer/uploader"?

**Comment by mnataraj92** [comment]
@jacobmpeters  done!

**Comment by jacobmpeters** [comment]
@mnataraj92 Thanks! I noticed that there are two NORC folks included on this test folder. I think I'd like to just test it with a folder with only a few Connect folks, so that we can make sure it's good-to-go before we share with NORC. What do you think?

**Comment by brotzmanmj** [comment]
After we have done that, we should also share a test file from dev or stage with NORC before running it in prod so they can confirm they can consume the format we provide.

**Comment by jeannewu** [comment]
@jacobmpeters @mnataraj92 would you like to test directly with sql  or R code? I will update and synch the R code to the git soon for the coming test.

**Comment by jeannewu** [comment]
@mnataraj92  in your previous message from the NORC feedback, the State of the participants' address would be included in the Birthday Card list, right?

**Comment by mnataraj92** [comment]
Hi all, 

I spoke with Michelle and Jake. @jacobmpeters I'll create a separate secure folder for the prod testing and will let you know when that is ready. 

While you're testing that, Michelle and I can check the stage data once the frequency counts Michelle requested are generated. @jeannewu can you please generate the frequency counts that Michelle requested above? Once that is complete, please let me know, and Michelle and I will check the data to confirm accuracy. Jake mentioned that the most recent stage file currently has an extra column that he will need to remove with the full name and address in one string, but that should be fixed tomorrow. 

We can then share a test file with NORC from stage to make sure everything looks good to them.

**Comment by jeannewu** [comment]
@brotzmanmj @mnataraj92 based on today's data, the NORC verified participants are counted in each database: prod, stg, dev listed<div _ngcontent-ng-c3480728048="" sandboxuid="0" style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: inherit; font-variant-caps: normal; font-weight: inherit; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none; box-sizing: border-box; outline: 0px; display: flex; -webkit-box-align: center; align-items: center;"><br class="Apple-interchange-newline"><button _ngcontent-ng-c3480728048="" mat-icon-button="" cfctooltip="Open sort menu" cfcdisabletooltip="Sorting not supported for REPEATED schema columns" class="bq-sort-menu-button cfc-button-small mdc-icon-button mat-mdc-icon-button gmat-mdc-button-with-prefix mat-unthemed mat-mdc-button-base gmat-mdc-button cm-button cfc-tooltip cfc-tooltip-disable-user-select-on-touch-device ng-star-inserted" mat-ripple-loader-uninitialized="" mat-ripple-loader-class-name="mat-mdc-button-ripple" mat-ripple-loader-centered="" jslog="65831;track:generic_click" aria-expanded="false" sandboxuid="0" id="_0rif_bq-results-table-header-menu-button-goog_1900590132" aria-haspopup="menu" aria-labelledby="_0rif_bq-results-table-header-menu-button-goog_1900590132 _0rif_cfc-labelledby-message-goog_1900590143" cfc-labelledby-host="" aria-describedby="_0rif_cdk-describedby-message-ng-1-23" cdk-describedby-host="ng-1"></button></div><div sandboxuid="0" style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: inherit; font-variant-caps: normal; font-weight: inherit; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none; box-sizing: border-box; outline: 0px; text-overflow: ellipsis;"></div><div sandboxuid="0" style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: inherit; font-variant-caps: normal; font-weight: inherit; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none; box-sizing: border-box; outline: 0px; text-overflow: ellipsis;"></div><div sandboxuid="0" style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: inherit; font-variant-caps: normal; font-weight: inherit; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none; box-sizing: border-box; outline: 0px; text-overflow: ellipsis;"></div><div sandboxuid="0" style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: inherit; font-variant-caps: normal; font-weight: inherit; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none; box-sizing: border-box; outline: 0px; text-overflow: ellipsis;"></div><div sandboxuid="0" style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: inherit; font-variant-caps: normal; font-weight: inherit; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none; box-sizing: border-box; outline: 0px; text-overflow: ellipsis;"></div><div sandboxuid="0" style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: inherit; font-variant-caps: normal; font-weight: inherit; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none; box-sizing: border-box; outline: 0px; text-overflow: ellipsis;"></div><div sandboxuid="0" style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: inherit; font-variant-caps: normal; font-weight: inherit; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none; box-sizing: border-box; outline: 0px; text-overflow: ellipsis;"></div><div sandboxuid="0" style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: inherit; font-variant-caps: normal; font-weight: inherit; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none; box-sizing: border-box; outline: 0px; text-overflow: ellipsis;"></div><div sandboxuid="0" style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: inherit; font-variant-caps: normal; font-weight: inherit; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none; box-sizing: border-box; outline: 0px; text-overflow: ellipsis;"></div><div sandboxuid="0" style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: inherit; font-variant-caps: normal; font-weight: inherit; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none; box-sizing: border-box; outline: 0px; text-overflow: ellipsis;"></div><div sandboxuid="0" style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: inherit; font-variant-caps: normal; font-weight: inherit; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none; box-sizing: border-box; outline: 0px; text-overflow: ellipsis;"></div><div sandboxuid="0" style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: inherit; font-variant-caps: normal; font-weight: inherit; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none; box-sizing: border-box; outline: 0px; text-overflow: ellipsis;"></div><div sandboxuid="0" style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: inherit; font-variant-caps: normal; font-weight: inherit; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none; box-sizing: border-box; outline: 0px; text-overflow: ellipsis;"></div><div sandboxuid="0" style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: inherit; font-variant-caps: normal; font-weight: inherit; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none; box-sizing: border-box; outline: 0px; text-overflow: ellipsis;"></div><div sandboxuid="0" style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: inherit; font-variant-caps: normal; font-weight: inherit; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none; box-sizing: border-box; outline: 0px; text-overflow: ellipsis;"></div><div sandboxuid="0" style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: inherit; font-variant-caps: normal; font-weight: inherit; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none; box-sizing: border-box; outline: 0px; text-overflow: ellipsis;"></div><div sandboxuid="0" style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: inherit; font-variant-caps: normal; font-weight: inherit; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none; box-sizing: border-box; outline: 0px; text-overflow: ellipsis;"></div><div sandboxuid="0" style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: inherit; font-variant-caps: normal; font-weight: inherit; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none; box-sizing: border-box; outline: 0px; text-overflow: ellipsis;"></div><div sandboxuid="0" style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: inherit; font-variant-caps: normal; font-weight: inherit; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none; box-sizing: border-box; outline: 0px; text-overflow: ellipsis;"></div><div sandboxuid="0" style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: inherit; font-variant-caps: normal; font-weight: inherit; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none; box-sizing: border-box; outline: 0px; text-overflow: ellipsis;"></div><div sandboxuid="0" style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: inherit; font-variant-caps: normal; font-weight: inherit; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none; box-sizing: border-box; outline: 0px; text-overflow: ellipsis;"></div><div sandboxuid="0" style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: inherit; font-variant-caps: normal; font-weight: inherit; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none; box-sizing: border-box; outline: 0px; text-overflow: ellipsis;"></div><div sandboxuid="0" style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: inherit; font-variant-caps: normal; font-weight: inherit; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none; box-sizing: border-box; outline: 0px; text-overflow: ellipsis;"></div><div sandboxuid="0" style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: inherit; font-variant-caps: normal; font-weight: inherit; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none; box-sizing: border-box; outline: 0px; text-overflow: ellipsis;"></div><div sandboxuid="0" style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: inherit; font-variant-caps: normal; font-weight: inherit; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none; box-sizing: border-box; outline: 0px; text-overflow: ellipsis;"></div><div sandboxuid="0" style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: inherit; font-variant-caps: normal; font-weight: inherit; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none; box-sizing: border-box; outline: 0px; text-overflow: ellipsis;"></div><div _ngcontent-ng-c3480728048="" sandboxuid="0" style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: inherit; font-variant-caps: normal; font-weight: inherit; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none; box-sizing: border-box; outline: 0px; display: flex; -webkit-box-align: center; align-items: center;"><button _ngcontent-ng-c3480728048="" mat-icon-button="" cfctooltip="Open sort menu" cfcdisabletooltip="Sorting not supported for REPEATED schema columns" class="bq-sort-menu-button cfc-button-small mdc-icon-button mat-mdc-icon-button gmat-mdc-button-with-prefix mat-unthemed mat-mdc-button-base gmat-mdc-button cm-button cfc-tooltip cfc-tooltip-disable-user-select-on-touch-device ng-star-inserted" mat-ripple-loader-class-name="mat-mdc-button-ripple" mat-ripple-loader-centered="" jslog="65831;track:generic_click" aria-expanded="false" sandboxuid="0" id="_0rif_bq-results-table-header-menu-button-goog_1900590431" aria-haspopup="menu" aria-labelledby="_0rif_bq-results-table-header-menu-button-goog_1900590431 _0rif_cfc-labelledby-message-goog_1900590143" cfc-labelledby-host="" aria-describedby="_0rif_cdk-describedby-message-ng-1-22" cdk-describedby-host="ng-1"></button></div><div _ngcontent-ng-c3480728048="" sandboxuid="0" style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: inherit; font-variant-caps: normal; font-weight: inherit; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none; box-sizing: border-box; outline: 0px; display: flex; -webkit-box-align: center; align-items: center;"><button _ngcontent-ng-c3480728048="" mat-icon-button="" cfctooltip="Open sort menu" cfcdisabletooltip="Sorting not supported for REPEATED schema columns" class="bq-sort-menu-button cfc-button-small mdc-icon-button mat-mdc-icon-button gmat-mdc-button-with-prefix mat-unthemed mat-mdc-button-base gmat-mdc-button cm-button cfc-tooltip cfc-tooltip-disable-user-select-on-touch-device ng-star-inserted" mat-ripple-loader-uninitialized="" mat-ripple-loader-class-name="mat-mdc-button-ripple" mat-ripple-loader-centered="" jslog="65831;track:generic_click" aria-expanded="false" sandboxuid="0" id="_0rif_bq-results-table-header-menu-button-goog_1900590433" aria-haspopup="menu" aria-labelledby="_0rif_bq-results-table-header-menu-button-goog_1900590433 _0rif_cfc-labelledby-message-goog_1900590143" cfc-labelledby-host="" aria-describedby="_0rif_cdk-describedby-message-ng-1-23" cdk-describedby-host="ng-1"></button></div><div sandboxuid="0" style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: inherit; font-variant-caps: normal; font-weight: inherit; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none; box-sizing: border-box; outline: 0px; text-overflow: ellipsis;"></div><div sandboxuid="0" style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: inherit; font-variant-caps: normal; font-weight: inherit; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none; box-sizing: border-box; outline: 0px; text-overflow: ellipsis;"></div><div sandboxuid="0" style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: inherit; font-variant-caps: normal; font-weight: inherit; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none; box-sizing: border-box; outline: 0px; text-overflow: ellipsis;"></div><div sandboxuid="0" style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: inherit; font-variant-caps: normal; font-weight: inherit; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none; box-sizing: border-box; outline: 0px; text-overflow: ellipsis;"></div><div sandboxuid="0" style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: inherit; font-variant-caps: normal; font-weight: inherit; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none; box-sizing: border-box; outline: 0px; text-overflow: ellipsis;"></div><div sandboxuid="0" style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: inherit; font-variant-caps: normal; font-weight: inherit; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none; box-sizing: border-box; outline: 0px; text-overflow: ellipsis;"></div><div sandboxuid="0" style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: inherit; font-variant-caps: normal; font-weight: inherit; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none; box-sizing: border-box; outline: 0px; text-overflow: ellipsis;"></div><div sandboxuid="0" style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: inherit; font-variant-caps: normal; font-weight: inherit; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none; box-sizing: border-box; outline: 0px; text-overflow: ellipsis;"></div><div sandboxuid="0" style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: inherit; font-variant-caps: normal; font-weight: inherit; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none; box-sizing: border-box; outline: 0px; text-overflow: ellipsis;"></div><div sandboxuid="0" style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: inherit; font-variant-caps: normal; font-weight: inherit; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none; box-sizing: border-box; outline: 0px; text-overflow: ellipsis;"></div><div sandboxuid="0" style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: inherit; font-variant-caps: normal; font-weight: inherit; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none; box-sizing: border-box; outline: 0px; text-overflow: ellipsis;"></div><div sandboxuid="0" style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: inherit; font-variant-caps: normal; font-weight: inherit; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none; box-sizing: border-box; outline: 0px; text-overflow: ellipsis;"></div><div sandboxuid="0" style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: inherit; font-variant-caps: normal; font-weight: inherit; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none; box-sizing: border-box; outline: 0px; text-overflow: ellipsis;"></div><div sandboxuid="0" style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: inherit; font-variant-caps: normal; font-weight: inherit; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none; box-sizing: border-box; outline: 0px; text-overflow: ellipsis;"></div><div sandboxuid="0" style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: inherit; font-variant-caps: normal; font-weight: inherit; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none; box-sizing: border-box; outline: 0px; text-overflow: ellipsis;"></div><div sandboxuid="0" style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: inherit; font-variant-caps: normal; font-weight: inherit; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none; box-sizing: border-box; outline: 0px; text-overflow: ellipsis;"></div><div sandboxuid="0" style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: inherit; font-variant-caps: normal; font-weight: inherit; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none; box-sizing: border-box; outline: 0px; text-overflow: ellipsis;"></div><div sandboxuid="0" style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: inherit; font-variant-caps: normal; font-weight: inherit; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none; box-sizing: border-box; outline: 0px; text-overflow: ellipsis;"></div><div sandboxuid="0" style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: inherit; font-variant-caps: normal; font-weight: inherit; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none; box-sizing: border-box; outline: 0px; text-overflow: ellipsis;"></div><div sandboxuid="0" style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: inherit; font-variant-caps: normal; font-weight: inherit; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none; box-sizing: border-box; outline: 0px; text-overflow: ellipsis;"></div><div sandboxuid="0" style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: inherit; font-variant-caps: normal; font-weight: inherit; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none; box-sizing: border-box; outline: 0px; text-overflow: ellipsis;"></div><div sandboxuid="0" style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: inherit; font-variant-caps: normal; font-weight: inherit; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none; box-sizing: border-box; outline: 0px; text-overflow: ellipsis;"></div><div sandboxuid="0" style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: inherit; font-variant-caps: normal; font-weight: inherit; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none; box-sizing: border-box; outline: 0px; text-overflow: ellipsis;"></div><div sandboxuid="0" style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: inherit; font-variant-caps: normal; font-weight: inherit; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none; box-sizing: border-box; outline: 0px; text-overflow: ellipsis;"></div><div sandboxuid="0" style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: inherit; font-variant-caps: normal; font-weight: inherit; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none; box-sizing: border-box; outline: 0px; text-overflow: ellipsis;"></div><div sandboxuid="0" style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: inherit; font-variant-caps: normal; font-weight: inherit; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none; box-sizing: border-box; outline: 0px; text-overflow: ellipsis;"></div><div sandboxuid="0" style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: inherit; font-variant-caps: normal; font-weight: inherit; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none; box-sizing: border-box; outline: 0px; text-overflow: ellipsis;"></div><div sandboxuid="0" style="font-style: normal; font-variant-caps: normal; font-weight: inherit; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none; box-sizing: border-box; outline: 0px; text-overflow: ellipsis; color: var(--cm-sys-color-state-disabled,rgba(0,0,0,0.54)) !important; fill: var(--cm-sys-color-state-disabled,rgba(0,0,0,0.54)) !important;"></div>
month (d_564964481) | birth_ID-prod | birth_ID_stage | birth_ID_dev
-- | -- | -- | --
1 | 1350 | 62 | 104
2 | 1320 | 26 | 48
3 | 1462 | 38 | 51
4 | 1393 | 34 | 47
5 | 1479 | 34 | 32
6 | 1526 | 34 | 36
7 | 1562 | 38 | 38
8 | 1538 | 42 | 31
9 | 1623 | 41 | 31
10 | 1516 | 33 | 33
11 | 1325 | 34 | 42
12 | 1508 | 17 | 14
month code error | 1 | 0 | 0
total Birthday (verified, with consent, not deceased from NORC) | 17603 | 433 | 507
deceased (overall) | 2 (NORC) | 1 (NORC) | 9 (1 Uoc, 8 NORC including NIH)
consent withdrawls (overall) | 253 (6 UoC, 247 NORC) | 20 (15 NORC + 5 UoC) | 30 (1 UoC, 29 NORC + NIH)

Hope the counts of Birthday in NORC undeceived, verified participants without consent withdrawals make sense.

**Comment by jacobmpeters** [comment]
This file in stage complies with NORCS formatting requests and is ready to be checked. https://nih.app.box.com/file/1381312677259

The following filters were applied:

```SQL
WHERE                                        # FILTERS:
  d_821247024 = '197316935'                  # Verifiation Status must be Verified
  AND birth_month = next_month_as_integer    # Birth Month must be Next Month
  AND d_747006172 != '353358909'             # Consent is NOT Withdrawn
  AND d_987563196 != '353358909'             # Participant is NOT Deceased
  AND d_827220437 != '809703864'             # Healthcare provider is NOT University of Chicago

**Comment by jacobmpeters** [comment]
@FrogGirl1123 Mentioned that University of Chicago should not be excluded, so I updated the filter.

This is the updated file: https://nih.app.box.com/file/1381327014282

```SQL
WHERE                                        # FILTERS:
  d_821247024 = '197316935'                  # Verifiation Status must be Verified
  AND birth_month = next_month_as_integer    # Birth Month must be Next Month
  AND d_747006172 != '353358909'             # Consent is NOT Withdrawn
  AND d_987563196 != '353358909'             # Participant is NOT Deceased
```

**Comment by brotzmanmj** [comment]
Just confirmed birth month you're using is 564964481?

**Comment by jacobmpeters** [comment]
@brotzmanmj Correct. I'll share the whole query in case you have more questions about CIDs.

```SQL
-- NORC Birthday Card List

-- Authors: Jing Wu and Jake Peters

-- Description: 
-- Get the Connect_ID, name and address of participants who have a birthday this month. 
-- Include only participants (1) that are verified, (2) that have not withdrawn consent, 
-- (3) that are not deceased and (4) that are not from University of Chicago.

# CSV_file_for_January_Birthday_Card_Recipients_2024

DECLARE next_month_as_string STRING DEFAULT FORMAT_DATE('%B', DATE_ADD(CURRENT_DATE, INTERVAL 1 MONTH));
DECLARE next_month_as_integer INT64 DEFAULT CAST(EXTRACT(month FROM DATE_ADD(CURRENT_DATE(), INTERVAL 1 MONTH)) AS INT64);

DECLARE uri_with_date STRING DEFAULT CONCAT(
  'gs://birthday_card_info/CSV_file_for_', next_month_as_string, '_Birthday_Card_Recipients_', FORMAT_DATE('%Y', CURRENT_DATE), '_boxfolder_237836684699_*.csv'
  );

EXPORT DATA OPTIONS (uri = CAST(uri_with_date AS STRING), format = 'CSV', overwrite = true, header = true, field_delimiter = ',') AS
WITH AddressInfo AS(
SELECT
  Connect_ID,
  token,
  CAST(d_564964481 AS INT64) AS birth_month,
  d_996038075 AS last_name,
  d_399159511 AS first_name,
  d_521824358 AS address_line_1,
  d_442166669 AS address_line_2,
  d_703385619 AS city,
  d_634434746 AS state,
  d_892050548 AS zip_code,
  d_821247024, 
  d_747006172, 
  d_987563196, 
  d_827220437
FROM
  `nih-nci-dceg-connect-stg-5519.FlatConnect.participants_JP`)
SELECT 
  -- Connect_ID,
  first_name,
  last_name,
  address_line_1,
  address_line_2,
  city,
  state,
  zip_code,
  -- CONCAT(first_name, ' ', last_name, '\n', 
  --        address_line_1, IFNULL(CONCAT('\n', address_line_2), ''), '\n',
  --        city, ', ', state, '  ', zip_code) 
  --        AS full_address_label,
FROM AddressInfo
WHERE                                        # FILTERS:
  d_821247024 = '197316935'                  # Verifiation Status must be Verified
  AND birth_month = next_month_as_integer    # Birth Month must be Next Month
  AND d_747006172 != '353358909'             # Consent is NOT Withdrawn
  AND d_987563196 != '353358909'             # Participant is NOT Deceased
;

```

**Comment by brotzmanmj** [comment]
Thanks for sharing!

**Comment by jacobmpeters** [comment]
@mnataraj92 @FrogGirl1123 

I mentioned that BigQuery chops the Birthday card data into many CSV files when the data are exported via scheduled query from Prod. I set up a Cloud Function to concatenate the files and export them to a specified folder in Box. 

The initial file has ben exported to a test folder set up by Madhuri (with only a few internal viewers).

https://nih.app.box.com/folder/238384827882

It is now ready to be independently checked @jeannewu , but @mnataraj92 will need to give you access first.

**Comment by mnataraj92** [comment]
I have given @jeannewu access

**Comment by jeannewu** [comment]
@jacobmpeters @mnataraj92 Thanks a lot. I have seen the birthday card data (file id =1387128776918) in the folder 238384827882. Just check This file is for the January Birthday Card (n=1618) of nine Connect Sites including UoC.

**Comment by mnataraj92** [comment]
Lauren from NORC mentioned a few additional columns she'd like added:

There are three fields we need added to the CSV file:
-	CONNECT_ID
o	This will not be included in any printed materials but will be needed when we return mailing outcomes.
-	PIN
o	This should be a copy of CONNECT_ID. It will be used by our case management to generate a random ID number that will be printed on the cards for tracking purposes.
-	DOBM
o	This should be a two digit field for birth month (with leading zeros for January - September). Even though the cases are already filtered by birth month prior to sharing, we will use this field to delete respondent information after the predetermined amount of time has passed.

**Comment by jacobmpeters** [comment]
@mnataraj92 I will add these back in and generate another test file. I'll let you know when this is done.

**Comment by jacobmpeters** [comment]
@mnataraj92 Just a clarification, PIN should just be the same value as CONNECT_ID, correct? 

This is a little confusing because the participants table has a 'pin' field already.. So, I just wanted to double check.

An alternative would be to just have an empty field called PIN where they can generate the random ID number for tracking purposes.

**Comment by mnataraj92** [comment]
hi @jacobmpeters I just emailed Lauren asking her preference (and cc'd you)

**Comment by mnataraj92** [comment]
@jacobmpeters Lauren mentioned that the “$PIN” field should have the same values as CONNECT_ID. It will be manipulated when they upload the files into their case management system.

**Comment by jacobmpeters** [comment]
The updated CSV file for stage and prod have been exported to the appropriate box folders.
Stage Testing: https://nih.app.box.com/file/1392302711509
Prod Testing: https://nih.app.box.com/file/1392309347706

**Comment by mnataraj92** [comment]
Thanks!! I'll ask Lauren to take a look at the stage file to make sure everything looks good to her

**Comment by jacobmpeters** [comment]
> The updated CSV file for stage and prod have been exported to the appropriate box folders. Stage Testing: https://nih.app.box.com/file/1392302711509 Prod Testing: https://nih.app.box.com/file/1392309347706

Oops. I have to add the leading for January - September

**Comment by jacobmpeters** [comment]
Leading zeros are now implemented for DOBM for both stage and prod.

**Comment by mnataraj92** [comment]
Lauren said the stage file looks good!

**Comment by jacobmpeters** [comment]
Great! After the punch list tomorrow, if everyone approves of the data, I'll point the delivery to the NORC folder so that it will deliver as scheduled on the 22nd of every month.

**Comment by jacobmpeters** [comment]
@mnataraj92 When you are able, can you add AutomationUser_1960507_dBBoRFHiim@boxdevedition.com
as a viewer/editor to https://nih.app.box.com/folder/237836526066?

I delivered the CSV file manually today. The PROD version of my Box app does not have uploader permissions to the NORC-facing app at the moment.

**Comment by jacobmpeters** [comment]
I alerted NORC that the CSV has been delivered to Box.

**Comment by mnataraj92** [comment]
@jacobmpeters done!

**Comment by sonyekere** [comment]
Marked done on punchlist.
